src/librustc_codegen_ironox/lib.rs

```diff
@@ -129,8 +135,40 @@ impl ExtraBackendMethods for IronOxCodegenBackend {
     }
 }
 
+#[derive(Debug)]
 pub struct ModuleIronOx {
-    asm: String
+    pub functions: Vec<IronOxFunction>,
+}
+
+impl ModuleIronOx {
+    pub fn new() -> ModuleIronOx {
+        ModuleIronOx {
+            functions: vec![],
+        }
+    }
+
+    pub fn add_function_with_type(
+        &mut self,
+        cx: &CodegenCx,
+        name: &str,
+        fn_type: Type) -> Value {
+        self.functions.push(IronOxFunction::new(cx, name, fn_type));
+        Value::Function(self.functions.len() - 1)
+    }
+
+    // XXX slow
+    pub fn get_function(&self, name: &str) -> Option<Value> {
```

Comment:
`XXX` will choke tests, use `FIXME`. Also it'd be better to reference functions by an index for constant time lookup. You don't have to fix the latter in this PR.

Reply:
I suppose this is dead code actually. I thought I needed to retrieve a function/global by name, which is why I ended up implementing this. I'll remove it altogether.